pmm/pmm2-api-tests.groovy

```diff
@@ -0,0 +1,98 @@
+void runStaging(String DOCKER_VERSION, CLIENT_VERSION, CLIENTS) {
+    stagingJob = build job: 'aws-staging-start', parameters: [
+        string(name: 'DOCKER_VERSION', value: DOCKER_VERSION),
+        string(name: 'CLIENT_VERSION', value: CLIENT_VERSION),
+        string(name: 'CLIENTS', value: CLIENTS),
+        string(name: 'NOTIFY', value: 'false'),
+        string(name: 'DAYS', value: '1')
+    ]
+    env.VM_IP = stagingJob.buildVariables.IP
+    env.VM_NAME = stagingJob.buildVariables.VM_NAME
+    env.PMM_URL = "http://${VM_IP}"
+}
+
+void destroyStaging(IP) {
+    build job: 'aws-staging-stop', parameters: [
+        string(name: 'VM', value: IP),
+    ]
+}
+
+pipeline {
+    agent {
+        label 'micro-amazon'
+    }
+    parameters {
+        string(
+            defaultValue: 'master',
+            description: 'Tag/Branch for pmm-api-tests repository',
+            name: 'GIT_BRANCH')
+        string(
+            defaultValue: 'perconalab/pmm-server:dev-latest',
+            description: 'PMM Server docker container version (image-name:version-tag)',
+            name: 'DOCKER_VERSION')
+        string(
+            defaultValue: 'dev-latest',
+            description: 'PMM Client version',
+            name: 'CLIENT_VERSION')
+    }
+    options {
+        skipDefaultCheckout()
+        disableConcurrentBuilds()
+    }
+    triggers {
+        upstream upstreamProjects: '', threshold: hudson.model.Result.SUCCESS
+    }
+
+    stages {
+        stage('Prepare') {
+            steps {
+                // clean up workspace and fetch pmm-qa repository
+                deleteDir()
+                git poll: false, branch: GIT_BRANCH, url: 'https://github.com/Percona-Lab/pmm-api-tests'
+                slackSend channel: '#pmm-ci', color: '#FFFF00', message: "[${JOB_NAME}]: build started - ${BUILD_URL}"
+            }
+        }
+        stage('Start staging') {
+            steps {
+                runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1')
+            }
+        }
+        stage('Sanity check') {
+            steps {
+                sh "curl --silent --insecure '${PMM_URL}/prometheus/targets' | grep localhost:9090"
```

Comment:
It is better to use `/ping` API – it returns 200 if PostgreSQL, Prometheus, etc. are up.